src/transformStateWithClones.js

```diff
@@ -7,7 +7,30 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const sumObject = [];
+  let newState = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(newState, action.extraData);
+        break;
+
+      case 'removeProperties':
+        action.keysToRemove.forEach((key) => {
+          delete newState[key];
+        });
+        break;
+
+      case 'clear':
+        newState = {};
+        break;
+    }
+
+    sumObject.push({ ...newState });
+  }
+
+  return sumObject;
```

Comment:
There is no error handling for the case when `actions` is not an array or contains elements that are not objects with a `type` property. It would be wise to add a guard clause at the beginning of the function to check the validity of the `actions` array and its elements.

Reply:
I think it will be in the future)